dart/dynamics/FreeJoint.cpp

```diff
@@ -575,6 +556,16 @@ const std::string& FreeJoint::getStaticType()
 }
 
 //==============================================================================
+void FreeJoint::setTransformFromChildBodyNode(const Eigen::Isometry3d& T)
+{
+  Joint::setTransformFromChildBodyNode(T);
+
+  // We update Jacobian here since the Jacobian of FreeJoint is a function of
+  // "T" but not of joint positions.
+  mJacobian = math::getAdTMatrix(T);
```

Comment:
The motivation of the ``_mandatory`` flag was because the function could be called either (1) when a property that the Jacobian depends on has changed or (2) when a state that the Jacobian *might* depend on has changed.

My impression with ``updateLocalJacobianDeriv()`` is that it's either always zero or always dependent on state. There wouldn't be a situation where it would depend on a property but not depend on a state. If I'm wrong about that, then we should definitely introduce a ``_mandatory`` flag for it. Although even if my assumption is correct, I would be okay with introducing a ``_mandatory`` flag for the sake of consistency if nothing else.